database/git_index.ksy

```diff
@@ -0,0 +1,150 @@
+meta:
+  id: git_index
+  title: Git index
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+  encoding: UTF-8
+doc-ref: https://github.com/git/git/blob/main/Documentation/technical/index-format.txt
+seq:
+  - id: header
+    type: header
+  - id: entries
+    type: entry
+    repeat: expr
+    repeat-expr: header.num_entries
+  - id: extensions
+    type: extension
+    repeat: until
+    repeat-until: _io.pos >= _io.size - len_hash
+  - id: checksum
+    size: len_hash
+instances:
+  len_hash:
+    value: 20   # SHA1, change to 32 for SHA256
+types:
+  header:
+    seq:
+      - id: magic
+        contents: "DIRC"
+      - id: version
+        type: u4
+        valid:
+          any-of: [2]    # only support version 2 now
+          #any-of: [2, 3, 4]
+      - id: num_entries
+        type: u4
+  entry:
+    seq:
+      - id: entry
+        type: entry_body
+      - id: padding
+        #size: (-(_io.pos - _root.header._sizeof) % 8)
+        size: padding_needed
+    instances:
+      padding_needed:
+        value: '(_io.pos - _root.header._sizeof) % 8 == 0 ? 8 : (-(_io.pos - _root.header._sizeof) % 8)'
```

Comment:
This could be simplified to this (borrowing from https://github.com/kaitai-io/kaitai_struct_formats/pull/409#discussion_r566748681):

```suggestion
        value: '8 - (_io.pos - _root.header._sizeof) % 8'
```

Although this would probably deserve a comment that inserting the 8-byte padding when it's already "aligned" is intentional, because it would normally look like a mistake.

Reply:
Graphs from Wolfram Alpha ([`(-x) mod 8`](https://www.wolframalpha.com/input?i2d=true&i=discontinuities+of+-x+mod+8), [`8 - (x mod 8)`](https://www.wolframalpha.com/input?i2d=true&i=discontinuities+of+8+-+%5C%2840%29x+mod+8%5C%2841%29)):

![-x_mod_8](https://user-images.githubusercontent.com/47499687/155282334-91fe0d6a-cb13-4e50-83ac-56c633d66c1c.gif)
![8-x_mod_8](https://user-images.githubusercontent.com/47499687/155282455-a6ef6cee-d74e-4dbf-a1de-3b3b4fce6399.gif)